{
  "Identifier": "CVE-2025-32381",
  "PackageSlug": "pypi/xgrammar",
  "Title": "xgrammar Vulnerable to Denial of Service (DoS) by abusing unbounded cache in memory",
  "Description": "Xgrammar includes a cache for compiled grammars to increase performance with repeated use of the same grammar. This cache is held in memory. Since the cache is unbounded, a system making use of xgrammar can be abused to fill up a host's memory and case a denial of service. For example, sending many small requests to an LLM inference server with unique JSON schemas would eventually cause this denial of service to occur.",
  "Date": "2025-04-09",
  "Pubdate": "2025-04-09",
  "AffectedRange": "\u003c0.1.18",
  "FixedVersions": [
    "0.1.18"
  ],
  "AffectedVersions": "All versions before 0.1.18",
  "NotImpacted": "All versions starting from 0.1.18",
  "Solution": "Upgrade to version 0.1.18 or above.",
  "Urls": [
    "https://nvd.nist.gov/vuln/detail/CVE-2025-32381",
    "https://github.com/advisories/GHSA-389x-67px-mjg3",
    "https://github.com/mlc-ai/xgrammar/security/advisories/GHSA-389x-67px-mjg3",
    "https://github.com/mlc-ai/xgrammar/pull/243",
    "https://github.com/vllm-project/vllm/pull/16283",
    "https://github.com/mlc-ai/xgrammar"
  ],
  "CvssV2": "",
  "CvssV3": "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H",
  "UUID": "6650dc47-9b49-45bc-9a6d-c060a4288f9f"
}